frappe/desk/query_report.py

```diff
@@ -191,29 +191,33 @@ def run(report_name, filters=None, user=None):
 
 def get_prepared_report_result(report, filters, dn="", user=None):
 	latest_report_data = {}
-	# Only look for completed prepared reports with given filters.
-	doc_list = frappe.get_all("Prepared Report",
-		filters={"status": "Completed", "report_name": report.name, "filters": filters, "owner": user})
-
 	doc = None
-	if len(doc_list):
-		if dn:
-			# Get specified dn
-			doc = frappe.get_doc("Prepared Report", dn)
-		else:
+	if dn:
+		# Get specified dn
+		doc = frappe.get_doc("Prepared Report", dn)
+	else:
+		# Only look for completed prepared reports with given filters.
+		doc_list = frappe.get_all("Prepared Report", filters={"status": "Completed", "filters": json.dumps(filters), "owner": user})
+		if doc_list:
 			# Get latest
 			doc = frappe.get_doc("Prepared Report", doc_list[0])
 
-		# Prepared Report data is stored in a GZip compressed JSON file
-		attached_file_name = frappe.db.get_value("File", {"attached_to_doctype": doc.doctype, "attached_to_name":doc.name}, "name")
-		compressed_content = get_file(attached_file_name)[1]
-		uncompressed_content = gzip_decompress(compressed_content)
-		data = json.loads(uncompressed_content)
-		if data:
-			latest_report_data = {
-				"columns": json.loads(doc.columns) if doc.columns else data[0],
-				"result": data
-			}
+	if doc:
+		try:
+			# Prepared Report data is stored in a GZip compressed JSON file
+			attached_file_name = frappe.db.get_value("File", {"attached_to_doctype": doc.doctype, "attached_to_name":doc.name}, "name")
+			compressed_content = get_file(attached_file_name)[1]
+			uncompressed_content = gzip_decompress(compressed_content)
+			data = json.loads(uncompressed_content)
+			if data:
+				latest_report_data = {
+					"columns": json.loads(doc.columns) if doc.columns else data[0],
+					"result": data
+				}
+		except Exception:
```

Comment:
I don't know if this is a good idea. You seem to be catching ALL exceptions in a large `try` block, which will make it harder to realize any bugs later, especially since this fails silently.

Based on the PR description, you seem to know where it'll break, so why not catch that specific exception and only handle that?

Reply:
This fails in the following cases (at least)

1. Prepared Report doesn't have any attached file
1. Prepared Report has more than one attached files
1. Attached file doc is available but the underlying file is missing from the filesystem.
1. The attached file is present but isn't a valid GZip file
1. Attached file isn't a valid JSON file
1. Attached file doesn't contain valid JSON content